File: leetcode/344. Reverse String/main.py
# Solved by @kitanoyoru
# https://leetcode.com/problems/reverse-string/

from typing import List


class Solution:
    def reverseString(self, s: List[str]) -> None:
        [start, end] = [0, len(s) - 1]

        while start < end:
            temp = s[start]
            s[start] = s[end]
            s[end] = temp
            start += 1
            end -= 1

File: leetcode/344. Reverse String/main.ts
// Solved by @kitanoyoru
// https://leetcode.com/problems/reverse-string/

/**
 Do not return anything, modify s in-place instead.
 */
const reverseString = (s: string[]): void => {
  let start = 0,
    end = s.length - 1,
    temp: string

  while (start < end) {
    temp = s[start]
    s[start] = s[end]
    s[end] = temp
    start++
    end--
  }
}
